src/modules/flake-parts/interface.nix

```
@@ -52,11 +31,27 @@ in {
       ({...}: {
         options = {
           dream2nix = {
+            instance = l.mkOption {
+              type = t.raw;
+              readOnly = true;
+              description = ''
+                The dream2nix instance.
+              '';
+            };
+            inputs = l.mkOption {
+              type = t.attrsOf t.attrs;
+              default = {};
+              description = ''
+                A list of inputs to generate outputs from.
+                Each one takes the same arguments `makeOutputs` takes.
+              '';
+            };
             outputs = l.mkOption {
-              type = t.attrsOf t.raw;
+              type = t.either (t.attrsOf t.raw) (t.attrsOf (t.attrsOf t.raw));
```

```suggestion
              type = t.lazyAttrsOf (t.lazyAttrsOf t.raw);
```

The either was only needed for the single `inputs` case, a consistent schema would be nicer; see earlier comment about 1 -> 2 -> 1.

`lazyAttrsOf` has better performance because for evaluating `foo.bar` it doesn't have to check all the other `foo.*` values just to produce a set of names that is then only queried for `bar`.